optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/entity/pillar/DefaultPillarSelector.java

```diff
@@ -252,22 +255,29 @@ public RandomSubPillarIterator() {
         @Override
         protected List<Object> createUpcomingSelection() {
             List<Object> basePillar = selectBasePillar();
+            int basePillarSize = basePillar.size();
+            if (basePillarSize == 1) { // no subpillar to select
+                return basePillar;
+            }
             // Known issue/compromise: Every subPillar should have same probability, but doesn't.
             // Instead, every subPillar size has the same probability.
-            int basePillarSize = basePillar.size();
-            int min = (minimumSubPillarSize > basePillarSize) ? basePillarSize : minimumSubPillarSize;
-            int max = (maximumSubPillarSize > basePillarSize) ? basePillarSize : maximumSubPillarSize;
+            int min = Math.min(minimumSubPillarSize, basePillarSize);
+            int max = Math.min(maximumSubPillarSize, basePillarSize);
```

Comment:
I replaced the original code with `Math.min()` so that the JVM can replace it with its intrinsic implementation, leading to an even more improved performance.